### PR TITLE
[Sqlvm] `az sql vm update`: Add configuration options for new SQL Assessment pre-requisites MMA->AMA migration

### PR DESCRIPTION
**Pull request by ai-abhinav**
az sql vm update --enable-assessment true -g <Resource Group Name> -n <SQL VM Name> --workspace-name <LA workspace's name> --workspace-rg <LA workspace's resource group name> --agent-rg <resource group for deploying AMA Agent dependent resources DCE and DCE> --workspace-sub <choosing an LA workspace outside the current VM subscription>

--workspace-sub is a new optional param. If specified this workspace is used, else the current workspace is picked as default
--agent-rg is a required parameter. It specifies the resource group in which the dependent resources get installed.
Note:* Considering making --agent-rg optional with a *warning* that agent resources will be deployed in the specified Resource Group but can be modified using --agent-rg.
--agent-rg is helpful because if respecified for another VM with the same LA workspace - the DCE, DCR get re-used and do not have to be re-created each time. 

Default value added to agent-rg
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
SQL VM

**Comment by yonzhan** [comment]
Please fix CI issues first.

File: src/azure-cli/azure/cli/command_modules/sqlvm/_params.py [hunk 1]
```diff
@@ -363,3 +363,7 @@ def load_arguments(self, _):
                    help='Name of the Log Analytics workspace to associate with VM.')
         c.argument('workspace_rg',
                    help='Resource group containing the Log Analytics workspace.')
+        c.argument('workspace_sub',
+                   help='Subscription containing the Log Analytics workspace.')
+        c.argument('agent_rg',
+                   help='Resource group where we provision the AMA Agent resources.')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sqlvm/_params.py:369]: Are these two parameters required? Adding required argument is breaking change which we need to pay much attention to

File: src/azure-cli/azure/cli/command_modules/sqlvm/custom.py [hunk 2]
```diff
@@ -475,57 +671,436 @@ def set_assessment_properties(cmd, instance, enable_assessment, enable_assessmen
...
+        # Raise error if workspace arguments not provided by user
+        if workspace_name is None or workspace_rg is None:
+            raise RequiredArgumentMissingError(
+                'Assessment requires a Log Analytics workspace and Log Analytics extension on VM - '
+                'workspace name and workspace resource group must be specified to deploy pre-requisites.')
+        if agent_rg is None:
+            raise RequiredArgumentMissingError(
+                'Assessment requires a Resource Group to deploy the AMA Agent resources- '
+                'Use the --agent-rg paramater to specify a resource group in this subscription. '
+                'It is recommended to reuse this resource group for other Assessment deployments using this Log Analytics workspace')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sqlvm/custom.py:684]: Seems like `agent_rg` is a newly added required argument, right?

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/sqlvm/custom.py]: Just noticed your PR description. Can you add a default value for `agent_rg` with warning message? We don't allow breaking change in this sprint. Our next breaking change window is Oct 2023

Comment (ai-abhinav) [reply src/azure-cli/azure/cli/command_modules/sqlvm/custom.py]: fixing CI and adding non breaking change default value for agent_rg. Opening new PR as this one was closed in merge attempt

**Comment by ai-abhinav** [comment]
Reopening due to CI issues in other PR

File: src/azure-cli/azure/cli/command_modules/sqlvm/_params.py (hunk 1 above)
Comment (ai-abhinav) [reply src/azure-cli/azure/cli/command_modules/sqlvm/_params.py]: both are made optional now.

**Review by evelyn-ys — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**